File: src/main/java/no/nav/sosialhjelp/soknad/web/rest/mappers/PersonMapper.java
package no.nav.sosialhjelp.soknad.web.rest.mappers;

import no.nav.sbl.soknadsosialhjelp.soknad.common.JsonNavn;
import no.nav.sosialhjelp.soknad.web.rest.ressurser.NavnFrontend;

public final class PersonMapper {

    private PersonMapper() {
    }

    public static String getPersonnummerFromFnr(String fnr){
        return fnr != null ? fnr.substring(6) : null;
    }

    public static JsonNavn mapToJsonNavn(NavnFrontend navn) {
        if (navn == null){
            return null;
        }
        return new JsonNavn()
                .withFornavn(navn.fornavn != null ? navn.fornavn : "")
                .withMellomnavn(navn.mellomnavn != null ? navn.mellomnavn : "")
                .withEtternavn(navn.etternavn != null ? navn.etternavn : "");
    }
}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/personalia/familie/PersonMapper.kt
package no.nav.sosialhjelp.soknad.personalia.familie

import no.nav.sbl.soknadsosialhjelp.soknad.common.JsonNavn
import no.nav.sosialhjelp.soknad.personalia.familie.dto.NavnFrontend
import org.apache.commons.lang3.StringUtils

object PersonMapper {

    fun getPersonnummerFromFnr(fnr: String?): String? {
        return fnr?.substring(6)
    }

    fun mapToJsonNavn(navn: NavnFrontend?): JsonNavn? {
        return if (navn == null) {
            null
        } else {
            JsonNavn()
                .withFornavn(navn.fornavn ?: "")
                .withMellomnavn(navn.mellomnavn ?: "")
                .withEtternavn(navn.etternavn ?: "")
        }
    }

    fun fulltNavn(navn: JsonNavn): String {
        val f = if (!StringUtils.isEmpty(navn.fornavn)) navn.fornavn else ""
        val m = if (!StringUtils.isEmpty(navn.mellomnavn)) " ${navn.mellomnavn}" else ""
        val e = if (!StringUtils.isEmpty(navn.etternavn)) " ${navn.etternavn}" else ""
        return (f + m + e).trim { it <= ' ' }
    }
}
